### ISSUE TITLE
[return-await] assertion return is throwing error when using always

### ISSUE THREAD

**Issue by climba03003** [typescript-eslint/typescript-eslint#2924]
I do not know if it is intended behavior or not. The return value is just a simple assertion and should not trigger the error.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/return-await": ["error", "always"]
  }
}
```

```TS
async function foo(): Promise<boolean> {
  const bar = await functionWithPromiseReturn()
  return bar !== null // error in here
}
```

**Expected Result**

I expect it should not having any error because it is just an simple assertion and should be treated as a value.

**Actual Result**
```
Returning an awaited value that is not a promise is not allowed.eslint@typescript-eslint/return-await
```

**Additional Info**
Current Workaround
```TS
async function foo(): Promise<boolean> {
  const bar = await functionWithPromiseReturn()
  const baz = bar !== null // it need to create another variable to make the error disappear
  return baz
}

async function foo(): Promise<boolean> {
  const bar = await functionWithPromiseReturn()
  return await (bar !== null) // await is useless in here
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.10.0` |
| `@typescript-eslint/parser`        | `4.10.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.16.0` |
| `node`                             | `v14.15.1` |

**Comment by bradzacher** [comment]
I'm unable to repro this against master.

![image](https://user-images.githubusercontent.com/7462525/103932743-27272680-50d7-11eb-98fa-cf429dca2721.png)

Please check your repro. If possible - provide your repro as a repository so it's perfectly reproducible.

**Comment by climba03003** [comment]
I will close this issue. After I clean all the cache of ```npm```, ```yarn``` and remove the ```node_modules```. Then reinstall the package again. The issue is gone. It should be my local cache issue
